arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -1544,11 +1538,31 @@ ::advance()
       case 0:
         // Success
         frame = std::move( new_frame );
+        if( frame_count )
+        {
+          ++( *frame_count );
+        }
 
         // Look up the dts of the packet that contained this frame
         if( auto const it = packet_pos_to_dts.find( frame->frame->pkt_pos );
             it != packet_pos_to_dts.end() )
         {
+          for( auto jt = raw_image_buffer.begin();
```

Comment:
This looks like more a while loop, why not use while instead?

Reply:
Doing it this way limits the scope of `jt` to only inside the loop. This would allow another loop at the same scope level to use the variable name `jt` when iterating over a different sequence. The reasoning is similar to why `it` is declared in the `if` statement instead of just before it: minimizing unnecessary variable scope.